svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -394,62 +394,76 @@ public function removeSemicolons($value)
+                }
+
+                let enableSlickFeature = 1;
+
+                const abGroupVal = `slk\${testName}\${targetPercentEnabled}`;
+                const storedOnOffVal = storage.getItem(abGroupVal);
+                const percentKey = `slickAbTestPercent-\${testName}`;
+                const storedPercentVal = parseInt(storage.getItem(percentKey));
+                
+                if (forceEnable !== true || !storedOnOffVal || storedPercentVal !== targetPercentEnabled) {
+                    enableSlickFeature = (Math.random() * 100) <= targetPercentEnabled ? 1 : 0;
+                }
+
+                storage.setItem(abGroupVal, enableSlickFeature);
+                storage.setItem(percentKey, targetPercentEnabled);
+
+                win.adthrive = win.adthrive || {};
+                win.adthrive.cmd = win.adthrive.cmd || [];
+                win.adthrive.cmd.push(() => { win.adthrive.config.abGroup.set(abGroupVal, enableSlickFeature); });
+
+                return !!enableSlickFeature;
+            };
+            JSBLOCK;
+
+            echo "<script>\n";
+            echo "'use strict';\n";
             if ($adThriveAbTest) {
-                echo '(() => {' . "\n";
-                echo '  window.adthrive = window.adthrive || {};' . "\n";
-                echo '  window.adthrive.cmd = window.adthrive.cmd || [];' . "\n";
-                echo '  let slickParams = new URLSearchParams(document.location.search.substring(1));' . "\n";
-                echo '  let slickAbParam = slickParams.get("abEnabled");' . "\n";
-                echo '  if (slickAbParam && ["on","off"].indexOf(slickAbParam) >= 0) {' . "\n";
-                echo '    window.adthrive_AB_enabled = slickAbParam;' . "\n";
-                echo '    if (window.localStorage) { window.localStorage.setItem("adthrive_AB_enabled", window.adthrive_AB_enabled); }' . "\n";
-                echo '  } else {' . "\n";
-                echo '    window.adthrive_AB_enabled = (window.localStorage ? window.localStorage.getItem("adthrive_AB_enabled") : undefined);' . "\n";
-                echo '    if (!window.adthrive_AB_enabled) {' . "\n";
-                echo '      window.adthrive_AB_enabled = Math.random() < ' . $adThriveAbFraction . ' ? "on" : "off";' . "\n";
-                echo '      if (window.localStorage) { window.localStorage.setItem("adthrive_AB_enabled", window.adthrive_AB_enabled); }' . "\n";
-                echo '    }' . "\n";
-                echo '  }' . "\n";
-                echo '  window.adthrive.cmd.push(function() {' . "\n";
-                echo '    window.adthrive.config.abGroup.set("slkstm", window.adthrive_AB_enabled);' . "\n";
-                echo '  });' . "\n";
-                echo '})();' . "\n";
-                echo 'if (window.adthrive_AB_enabled === "on") {' . "\n";
+                echo $jsBlock;
+                echo "if (window.slickAbTestResult(" + $enabledPercent + ")) {\n";
```

Comment:
replace + with .

Reply:
Good catch... it's been far too long since I've done any PHP!